### ISSUE TITLE
MetaMask takes an unreasonably long time to show tx confirmation

### ISSUE THREAD

**Issue by bwheeler96** [MetaMask/metamask-extension#5551]
**Describe the bug**
MetaMask takes a super long time to show confirmed transactions. Right now I have a transaction with 23 block confirmations on Etherscan that is still "pending" in MM.

**To Reproduce**
Make a transaction.

**Expected behavior**
Show tx confirmed after ~3 block confirmations

**Browser details (please complete the following information):**
 - OS: Max
 - Browser Brave
 - MetaMask Version [Latest]
 - New UI

**Comment by bwheeler96** [comment]
At the very least, having an indicator in the UI for "last block synced" and the number of seconds ago (preferably the block timestamp as an ultimate source of truth for when the last block actually was)

**Comment by guix77** [comment]
Same thing here (MM 4.16 + Rinkeby), was working lie a charm on 4.14.

Note: for me, switching accounts immediatly triggers the TX detection by MM

**Comment by guix77** [comment]
Seems pretty random to me: now it's working like before. Might be a network issue and maybe not from Metamask's fault. I'm finding Rinkeby a lot slower today, than it was for instance last friday. Maybe too many ppl joined Rinkeby during the Ropsten problems?

Anyways next time I encounter this pb I'll check against a local Ganache with which I get pretty consistent results (mining time set to 15s).

**Comment by levity** [comment]
Was seeing this bug too. From a comment on #5553 I got the idea to try switching to a different network and switching back. This makes it stop happening.

**Comment by chaitanyapotti** [comment]
We've been facing the same problem. As a dapp which uses metamask provider to return tx confirmation status, our buttons keep spinning indefinitely on the UI because of this issue. the confirmation is issued instantly in 4.14 but is broken in 4.16.

I've done enough testing to figure that the problem is with the metamask extension and not infura/rinkeby platform

Can someone fix it asap?

**Comment by REPTILEHAUS** [comment]
Also getting this issue. Lots of pending txns and now lots of pending txn cancellations. Any solution for this ?

Tried switching networks and accounts but they still are stuck. Running latest UI on chrome on m
OSX Mojave

**Comment by bdresser** [comment]
Thanks all - this was fixed in https://github.com/MetaMask/metamask-extension/pull/5614 and rolled out in 4.17.

If you experience this issue on >=4.17 please let me know and we'll re-open.

**Comment by REPTILEHAUS** [comment]
@bdresser Im running 4.17.1 and this is still an issue.. multiple transactions still in pending including the cancel requests and clicking view on etherscan link show "undefined"

**Comment by bdresser** [comment]
@REPTILEHAUS this will fix for future transactions, but won't clear out your history.

does your oldest transaction (lowest nonce) appear on Etherscan? If it does, and it's still pending on Etherscan, try submitting a Cancellation or Speed Up attempt now that you're on 4.17.1 and it should go through. 

If your oldest transaction doesn't appear on Etherscan it was possibly never broadcast, head to Settings and reset your account. This will clear your tx history and you should be all set.

**Comment by REPTILEHAUS** [comment]
Ive been on v4.17.1 a while now and this issue still persists. I reset my account and did another txn just there (gas 35/gas limit 225558) and still no luck

**Comment by bdresser** [comment]
@REPTILEHAUS "no luck" meaning the transaction never makes it to Etherscan? Is it still using nonce 0?

**Comment by REPTILEHAUS** [comment]
yes no luck as in it doesnt reach etherscan and yes the nonce is still 0

**Comment by bdresser** [comment]
sorry to be a drag, but could you open a separate issue please? and would you let us know if you see any errors in the UI console or background console when you submit a transaction?

**Comment by REPTILEHAUS** [comment]
Sure, submitted it here. I see no console output either but will keep testing. https://github.com/MetaMask/metamask-extension/issues/5687

**Comment by bwheeler96** [comment]
Closes issue then asks for same issue to be reopened as a separate issue.

Welcome to MetaMask everyone. Sorry for the salt.
